src/duplicacy_chunkdownloader.go

```diff
@@ -324,22 +324,39 @@ func (downloader *ChunkDownloader) Download(threadIndex int, task ChunkDownloadT
 		LOG_DEBUG("CHUNK_FOSSIL", "Chunk %s has been marked as a fossil", chunkID)
 	}
 
-	err = downloader.storage.DownloadFile(threadIndex, chunkPath, chunk)
-	if err != nil {
-		LOG_ERROR("UPLOAD_FATAL", "Failed to download the chunk %s: %v", chunkID, err)
-		return false
-	}
+	const MaxDownloadAttempts = 3
+	for downloadAttempt := 0;; downloadAttempt++ {
```

Comment:
This loop would be cleaner if you pull out a helper method that attempts to download, decrypt, and compare the chunk's ID, and returns true (plus any other needed return values, though I'm not seeing any from a quick look) if all those things succeeded and false if any of them failed. Then your loop's body could handle the logic of what to do if you've failed (checking if you're past the limit or not) in a single place.

That separates the concerns (attempting a single download vs. retrying 3 times), and also starts to carve up the 130-line method into something more reasonably-sized.